vector/src/main/java/im/vector/app/features/location/MapTilerMapView.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.util.AttributeSet
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolManager
+import com.mapbox.mapboxsdk.plugins.annotation.SymbolOptions
+import com.mapbox.mapboxsdk.style.layers.Property
+import im.vector.app.BuildConfig
+
+class MapTilerMapView @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : MapView(context, attrs, defStyleAttr), VectorMapView {
+
+    private var map: MapboxMap? = null
+    private var symbolManager: SymbolManager? = null
+    private var style: Style? = null
+
+    override fun initialize(listener: VectorMapListener) {
```

Comment:
Nit: maybe better to use a lambda, `listener` is a name rather used for global object listeners.

Reply:
Nice idea, done.